autogpt_platform/backend/backend/data/graph.py

```diff
@@ -47,13 +49,21 @@ def __hash__(self):
         return hash((self.source_id, self.sink_id, self.source_name, self.sink_name))
 
 
-class Node(BaseDbModel):
+class CreatableNode(BaseDbModel):
```

Comment:
nit: whats a creatable node

Reply:
do you mean it needs a comment or docstring explaining that?

Reply:
It isn’t clear what the difference between node and creatable node is and why it’s renamed  but doesn’t inherit from a presumably new Node. I’d expect Node to be a partner or parent of Creatable Node while this appears to be that parent which requires clarification in the form of a doc string or similar

Reply:
I have renamed those, there is no longer CretableNode. There is NodeModel, which basically just a model version of Node which has helper functions and caches some computed fields, whereas the Node is just a pydantic class with fields.

Reply:
> This requires a huge renaming on the existing main & test codes

I don't see that as an issue as long as it is an improvement.

About this rename: I'm not sure how to interpret `NodeModel` since `Node` is also a model. It seems very ambiguous to me. I have tried to come up with a more descriptive naming scheme but I just come back to `CreatableX` because that's what it is functionally.

The exception is `Node`, which can also be `Node` (without `graph_id`) <-> `GraphNode` (with `graph_id`).